source/extensions/filters/network/thrift_proxy/protocol.h

```diff
@@ -0,0 +1,191 @@
+#pragma once
+
+#include <memory>
+#include <stack>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/pure.h"
+
+#include "common/common/assert.h"
+#include "common/common/macros.h"
+#include "common/singleton/const_singleton.h"
+
+#include "absl/types/optional.h"
```

Comment:
Is this needed in `protocol.h`? It doesn't look like this file uses `absl::optional` (but I might have missed something).